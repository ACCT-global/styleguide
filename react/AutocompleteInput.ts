export { default } from './components/Autocomplete'
