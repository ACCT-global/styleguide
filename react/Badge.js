import Tag from './components/Tag/index'
import deprecated from './modules/deprecated'

export default deprecated(
  {
    useNewComponent: {
      Badge: 'Tag',
    },
  },
  Tag
)
