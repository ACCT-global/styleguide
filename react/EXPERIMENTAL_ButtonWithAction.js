import EXPERIMENTAL_ButtonWithAction from './components/EXPERIMENTAL_ButtonWithAction/index'

export default EXPERIMENTAL_ButtonWithAction
