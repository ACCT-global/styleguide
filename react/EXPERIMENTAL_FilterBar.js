import EXPERIMENTAL_FilterBar from './components/EXPERIMENTAL_FilterBar/index'

export default EXPERIMENTAL_FilterBar
