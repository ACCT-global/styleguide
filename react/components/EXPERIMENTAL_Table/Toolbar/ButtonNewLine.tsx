import React, { FC } from 'react'

import ActionMenu from '../../ActionMenu/index.js'
import ButtonGroup from '../../ButtonGroup/index.js'
import IconPlus from '../../icon/Plus/index.js'

import Button, { ButtonProps } from './Button'
import { ICON_SIZE } from '../constants'

const ButtonNewLine: FC<ButtonNewLineProps> = ({ actions, ...buttonProps }) => {
  return actions ? (
    <ButtonGroup
      buttons={[
        <Button
          isActiveOfGroup
          key="new-line-button"
          variation="primary"
          icon={<IconPlus solid size={ICON_SIZE.LIGHT} />}
          {...buttonProps}
        />,
        <ActionMenu
          isActiveOfGroup
          key="actions-button"
          buttonProps={buttonProps}
          options={actions}
        />,
      ]}
    />
  ) : (
    <Button
      icon={<IconPlus solid size={ICON_SIZE.LIGHT} />}
      variation="primary"
      {...buttonProps}
    />
  )
}

ButtonNewLine.defaultProps = {
  size: 'small',
}

type Action = {
  label: string
  onClick: Function
  toggle: {
    checked: boolean
    semantic: boolean
  }
}

export type ButtonNewLineProps = ButtonProps & {
  actions: Array<Action>
}

export default ButtonNewLine
