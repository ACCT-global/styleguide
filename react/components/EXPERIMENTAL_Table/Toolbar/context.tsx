import React, { FC, useContext, createContext } from 'react'

import { E2ETestable } from '../types'

const ToolbarContext = createContext<E2ETestable>(null)

export const ToolbarProvider: FC<E2ETestable> = ({ children, testId }) => {
  return (
    <ToolbarContext.Provider value={{ testId }}>
      {children}
    </ToolbarContext.Provider>
  )
}

export function useToolbarContext() {
  const context = useContext(ToolbarContext)
  if (!context) {
    throw new Error('Do not use Toolbar composites outside of context')
  }
  return context
}
