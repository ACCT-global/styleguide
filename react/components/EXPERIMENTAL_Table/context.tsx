import React, { FC, useContext, createContext } from 'react'

import { E2ETestable } from './types'
import useTableMeasures from './hooks/useTableMeasures'

type Measures = ReturnType<typeof useTableMeasures>

const TestingContext = createContext<E2ETestable>(null)
const MeasuresContext = createContext<Measures>(null)

type TableType = E2ETestable & {
  measures: Measures
}

export const TableProvider: FC<TableType> = ({
  children,
  testId,
  measures,
}) => {
  return (
    <TestingContext.Provider value={{ testId }}>
      <MeasuresContext.Provider value={measures}>
        {children}
      </MeasuresContext.Provider>
    </TestingContext.Provider>
  )
}

export function useTestingContext() {
  const context = useContext(TestingContext)
  if (!context) {
    throw new Error(
      'Do not use E2ETestable components outside of the TestingContext'
    )
  }
  return context
}

export function useMeasuresContext() {
  const context = useContext(MeasuresContext)
  if (!context) {
    throw new Error(
      'Do not use measurable components outside of the MeasuresContext'
    )
  }
  return context
}
