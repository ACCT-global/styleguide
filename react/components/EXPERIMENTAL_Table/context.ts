import { createContext } from 'react'

const Context = createContext<TableState>(null)
const { Provider } = Context

export default Context
export { Provider as TableProvider }
