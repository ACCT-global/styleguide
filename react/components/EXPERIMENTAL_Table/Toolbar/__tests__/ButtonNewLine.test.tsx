import React from 'react'
import { render } from '@testing-library/react'

import { ButtonGroupProvider } from '../context'
import ButtonNewLine from '../ButtonNewLine'

describe('Table V2 @ Toolbar/ButtonNewLine spec', () => {
  const basicProps = {
    label: 'New',
    onClick: () => {},
  }

  it('matches snapshot', () => {
    const button = render(
      <ButtonGroupProvider testId="button-group">
        <ButtonNewLine {...basicProps} />
      </ButtonGroupProvider>
    )
    expect(button.asFragment()).toMatchSnapshot()
  })
})
