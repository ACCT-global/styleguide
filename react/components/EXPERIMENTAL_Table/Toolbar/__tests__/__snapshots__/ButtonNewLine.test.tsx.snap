// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table V2 @ Toolbar/ButtonNewLine spec matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="relative"
    data-testid="button-group__new-line__container"
    id="vtex-table-v2__toolbar__button-newline__container"
  >
    <button
      class="vtex-button bw1 ba fw5 v-mid relative pa0 lh-solid br2 min-h-small t-action--small bg-action-primary b--action-primary c-on-action-primary hover-bg-action-primary hover-b--action-primary hover-c-on-action-primary pointer "
      data-testid="button-group__new-line"
      tabindex="0"
      type="button"
    >
      <div
        class="vtex-button__label flex items-center justify-center h-100 ph5 "
        style="padding-top: .25em; padding-bottom: .32em;"
      >
        <span
          class="flex items-center nr1"
        >
          <div
            class="mr2 nl2 flex items-center"
          >
            <span
              class=""
              style="margin-top: 1.5px;"
            >
              <svg
                class="vtex__icon-plus vtex__icon-plus--solid "
                fill="none"
                height="16"
                viewBox="0 0 20 20"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
                xmlns:xlink="http://www.w3.org/1999/xlink"
              >
                <path
                  clip-rule="evenodd"
                  d="M13.6569 13.6569C16.7811 10.5326 16.7811 5.46735 13.6569 2.34316C10.5327 -0.781032 5.46739 -0.781075 2.34316 2.34316C-0.781075 5.46739 -0.781032 10.5327 2.34316 13.6569C5.46735 16.7811 10.5326 16.7811 13.6569 13.6569ZM8.00006 6.58567L10.8285 3.75724L12.2427 5.17159L9.41423 8.00001L12.2427 10.8284L10.8285 12.2425L8.00001 9.41423L5.15726 12.257L3.74309 10.8426L6.58576 7.99997L3.74313 5.15734L5.15747 3.74309L8.00006 6.58567Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                  transform="translate(10.1006 -1.31372) rotate(45)"
                />
              </svg>
            </span>
          </div>
          <div>
            <span
              class=""
            >
              New
            </span>
          </div>
        </span>
      </div>
    </button>
  </div>
</DocumentFragment>
`;
